evm-contracts/src/v0.6/dev/BlockhashStore.sol

```diff
@@ -0,0 +1,67 @@
+pragma solidity 0.6.2;
+
+/**
+ * @title BlockhashStore
+ * @notice This contract provides a way to access blockhashes older than
+ * the 256 block limit imposed by the BLOCKHASH opcode.
+ * You may assume that any blockhash stored by the contract is correct.
+ */
+contract BlockhashStore {
+
+  mapping(uint => bytes32) internal s_blockhashes;
+
+  /**
+   * @notice stores blockhash of a given block, assuming it is available through BLOCKHASH
+   * @param n the number of the block whose blockhash should be stored
+   */
+  function store(uint256 n) public {
+    bytes32 h = blockhash(n);
+    require(h != 0x0, "blockhash(number) failed");
+    s_blockhashes[n] = h;
+  }
+
+
+  /**
+   * @notice stores blockhash of the earliest block still available through BLOCKHASH.
```

Comment:
How will this be used?

Reply:
Just a convenience function. Imagine you want to store the earliest blockhash that's still cheaply available through BLOCKHASH. 
If you use the regular `store()` function, you have to guess in what block your tx will be included.

Reply:
:+1: